File: ios/Sources/ios/Business/Internal/UseCases.swift
/// Single functional piece of business logic.
///
/// Input data comes from DI.
///
/// - Parameter OutputType: Use case produced data type
public typealias UseCase<OutputType> = () -> OutputType

/// Single functional piece of business logic that reacts to input data.
///
/// Input data comes from DI.
///
/// - Parameter InputType: Input data type
/// - Parameter OutputType: Use case produced data type
public typealias InteractionUseCase<InputType, OutputType> = (InputType) -> OutputType

File: android/app/src/main/kotlin/com/test/business/UseCases.kt
package com.test.business

/**
 * Single functional piece of business logic.
 *
 * Input data comes from DI.
 *
 * @param OutputType Use case produced data type
 */
interface UseCase<OutputType> : () -> OutputType

/**
 * Single functional piece of business logic that reacts to input data.
 *
 * @param InputType Input data type
 * @param OutputType Use case produced data type
 */
interface InteractionUseCase<InputType, OutputType> : (InputType) -> OutputType
